src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1797,22 +1815,45 @@ private static void ThrowOverflowOrFormatException(bool overflow, string overflo
                (Exception)new FormatException(SR.Format_InvalidString);
         }
 
-        private static bool TryNumberToDouble(ref NumberBuffer number, ref double value)
+        private static bool TryNumberToDouble(ref NumberBuffer number, out double value)
         {
-            double d = NumberToDouble(ref number);
-            if (!double.IsFinite(d))
+            ulong bits = NumberToFloatingPointBitsRoslyn(ref number, in FloatingPointInfo.Double);
+            double result = BitConverter.Int64BitsToDouble((long)(bits));
+
+            if (!double.IsFinite(result))
```

Comment:
Our current behavior is to return `false` and `0` for overflow. However, the correct behavior is to return `Infinity` with the appropriate sign.

Fixing this would fall into the realm of breaking changes and we would need to determine whether this counts as a parsing failure (which would return `false` from `TryParse` and would throw for `Parse`).

Reply:
Under the default IEEE rules, this would signal the `overflow exception` (if enabled); However, we explicitly disable and do not support IEEE exception handling; which makes the current behavior a bit of an oddity.

Reply:
CC. @jkotas, do you have an opinion here on the correct behavior?

I believe the options are:
* Keep the current behavior and return false, setting `result` to 0
* Return false, but set the result to `Infinity` with the appropriate sign
  * This would throw for `Parse`, but for `TryParse` would allow interested users to differentiate between "Format" and "Overflow"; it should not cause any difference in behavior for existing code paths
* Return true and set the result to `Infinity` with the appropriate sign
  * Both `Parse` and `TryParse` would work and this is arguably the most correct. However, some code may not know to handle the overflow case correctly

Reply:
I've selected to `return true` and set the result to `Infinity` with the appropriate sign, for now. It is the most correct behavior and is what other languages/frameworks do when IEEE exception handling is disabled.